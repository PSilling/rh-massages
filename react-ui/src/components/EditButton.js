// react imports
import React, { Component } from 'react';

/**
 * Custom edit button component.
 */
class EditButton extends Component {

  render() {
    return(
      <span>
        <button style={{ 'color': '#000' }}  type="button" className="btn btn-link" onClick={this.props.onEdit}>
          <span className="glyphicon glyphicon-pencil"></span>
        </button>
      </span>
    );
  }
}

export default EditButton
