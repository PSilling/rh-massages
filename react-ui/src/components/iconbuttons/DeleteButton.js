// react imports
import React, { Component } from 'react';

// component imports
import ConfirmationModal from '../modals/ConfirmationModal';

// util imports
import _t from '../../util/Translations';

/**
 * Custom delete button component.
 */
class DeleteButton extends Component {

  state = {active: false}

  handleToggle = () => {
    this.setState({active: !this.state.active});
  }

  render() {
    return(
      <span>
        <button style={{ 'color': '#000' }} type="button" className="btn btn-link"
          onClick={this.handleToggle} title={ _t.translate("Delete") }>
          <span className="glyphicon glyphicon-remove"></span>
        </button>
        {this.state.active ?
          <ConfirmationModal
            message={ _t.translate('Are you sure? This action cannot be reverted.') }
            onClose={() => this.handleToggle()}
            onConfirm={() => {
              this.handleToggle();
              this.props.onDelete();
            }}
          /> : ''
        }
      </span>
    )
  }
}

export default DeleteButton
