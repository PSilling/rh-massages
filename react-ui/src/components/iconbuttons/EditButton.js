// react imports
import React from "react";
import PropTypes from "prop-types";

// util imports
import _t from "../../util/Translations";

/**
 * Icon only button used for element editing.
 */
const EditButton = function EditButton(props) {
  return (
    <span>
      <button
        style={{ color: "#000" }}
        type="button"
        className="btn btn-link"
        onClick={props.onEdit}
        title={_t.translate("Edit")}
      >
        <span className="glyphicon glyphicon-pencil" />
      </button>
    </span>
  );
};

EditButton.propTypes = {
  /** function to be called on button click */
  onEdit: PropTypes.func.isRequired
};

export default EditButton;
