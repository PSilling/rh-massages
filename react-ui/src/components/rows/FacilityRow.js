// react imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// component imports
import DeleteButton from '../iconbuttons/DeleteButton';
import EditButton from '../iconbuttons/EditButton';

/**
 * Custom table row for a facility.
 */
class FacilityRow extends Component {

  shouldComponentUpdate(nextProps) {
    return (this.props.facility.id !== nextProps.facility.id
            || this.props.facility.name !== nextProps.facility.name);
  }

  render() {
    return (
      <tr>
        <td>{this.props.facility.name}</td>
        <td width="105px">
          <span className="pull-right">
            <span style={{ 'marginRight': '5px' }}>
              <EditButton onEdit={this.props.onEdit} />
            </span>
            <DeleteButton onDelete={this.props.onDelete} />
          </span>
        </td>
      </tr>
    );
  }
}

FacilityRow.propTypes = {
  facility: PropTypes.object.isRequired, // the facility for this row
  onEdit: PropTypes.func.isRequired, // function called on Facility edit
  onDelete: PropTypes.func.isRequired, // function called on Facility delete
}

export default FacilityRow
