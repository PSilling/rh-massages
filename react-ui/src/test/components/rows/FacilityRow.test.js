// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import DeleteButton from '../../../components/iconbuttons/DeleteButton';
import EditButton from '../../../components/iconbuttons/EditButton';
import FacilityRow from '../../../components/rows/FacilityRow';

test('renders content with correct props', () => {
  const testEditFunction = jest.fn(),
        testDeleteFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <FacilityRow facility={{ name: "test" }} onEdit={testEditFunction}
            onDelete={testDeleteFunction} />
        ),
        testInstance = testRenderer.root;

  let cells = testInstance.findAllByType('td'),
      editButton = testInstance.findByType(EditButton),
      deleteButton = testInstance.findByType(DeleteButton),
      treeJSON = testRenderer.toJSON();

  expect(cells[0].props.children).toEqual("test");
  expect(editButton.props.onEdit).toBe(testEditFunction);
  expect(deleteButton.props.onDelete).toBe(testDeleteFunction);
  expect(treeJSON).toMatchSnapshot();
});
