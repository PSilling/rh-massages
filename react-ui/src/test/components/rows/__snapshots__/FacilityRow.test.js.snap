// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FacilityRow
    facility={
      Object {
        "name": "test",
      }
    }
    onDelete={[MockFunction]}
    onEdit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          test
        </td>,
        <td
          width="105px"
        >
          <span
            className="float-right"
          >
            <TooltipIconButton
              disabled={false}
              icon="edit"
              onClick={[MockFunction]}
              size="sm"
              style={
                Object {
                  "border": "0px solid transparent",
                }
              }
              tooltip="Edit"
            />
            <ConfirmationIconButton
              className="ml-2"
              dialogMessage="Are you sure? This action cannot be reverted."
              icon="trash"
              onConfirm={[MockFunction]}
              tooltip="Delete"
            />
          </span>
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="float-right"
          >
            <TooltipIconButton
              disabled={false}
              icon="edit"
              onClick={[MockFunction]}
              size="sm"
              style={
                Object {
                  "border": "0px solid transparent",
                }
              }
              tooltip="Edit"
            />
            <ConfirmationIconButton
              className="ml-2"
              dialogMessage="Are you sure? This action cannot be reverted."
              icon="trash"
              onConfirm={[MockFunction]}
              tooltip="Delete"
            />
          </span>,
          "width": "105px",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TooltipIconButton
                disabled={false}
                icon="edit"
                onClick={[MockFunction]}
                size="sm"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Edit"
              />,
              <ConfirmationIconButton
                className="ml-2"
                dialogMessage="Are you sure? This action cannot be reverted."
                icon="trash"
                onConfirm={[MockFunction]}
                tooltip="Delete"
              />,
            ],
            "className": "float-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "icon": "edit",
                "onClick": [MockFunction],
                "size": "sm",
                "style": Object {
                  "border": "0px solid transparent",
                },
                "tooltip": "Edit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "ml-2",
                "dialogMessage": "Are you sure? This action cannot be reverted.",
                "icon": "trash",
                "onConfirm": [MockFunction],
                "tooltip": "Delete",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "span",
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            test
          </td>,
          <td
            width="105px"
          >
            <span
              className="float-right"
            >
              <TooltipIconButton
                disabled={false}
                icon="edit"
                onClick={[MockFunction]}
                size="sm"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Edit"
              />
              <ConfirmationIconButton
                className="ml-2"
                dialogMessage="Are you sure? This action cannot be reverted."
                icon="trash"
                onConfirm={[MockFunction]}
                tooltip="Delete"
              />
            </span>
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="float-right"
            >
              <TooltipIconButton
                disabled={false}
                icon="edit"
                onClick={[MockFunction]}
                size="sm"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Edit"
              />
              <ConfirmationIconButton
                className="ml-2"
                dialogMessage="Are you sure? This action cannot be reverted."
                icon="trash"
                onConfirm={[MockFunction]}
                tooltip="Delete"
              />
            </span>,
            "width": "105px",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TooltipIconButton
                  disabled={false}
                  icon="edit"
                  onClick={[MockFunction]}
                  size="sm"
                  style={
                    Object {
                      "border": "0px solid transparent",
                    }
                  }
                  tooltip="Edit"
                />,
                <ConfirmationIconButton
                  className="ml-2"
                  dialogMessage="Are you sure? This action cannot be reverted."
                  icon="trash"
                  onConfirm={[MockFunction]}
                  tooltip="Delete"
                />,
              ],
              "className": "float-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "icon": "edit",
                  "onClick": [MockFunction],
                  "size": "sm",
                  "style": Object {
                    "border": "0px solid transparent",
                  },
                  "tooltip": "Edit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "ml-2",
                  "dialogMessage": "Are you sure? This action cannot be reverted.",
                  "icon": "trash",
                  "onConfirm": [MockFunction],
                  "tooltip": "Delete",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "span",
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
